modules/aws/src/main/scala/com/snowplowanalytics/aws/sqs/SQS.scala

```diff
@@ -0,0 +1,97 @@
+package com.snowplowanalytics.aws.sqs
+
+import scala.concurrent.duration._
+import scala.collection.JavaConverters._
+
+import cats.effect.{Timer, Resource, Sync}
+import cats.implicits._
+
+import fs2.Stream
+
+import software.amazon.awssdk.services.sqs.{SqsClient, SqsClientBuilder}
+import software.amazon.awssdk.services.sqs.model.{GetQueueUrlRequest, DeleteMessageRequest, SqsException, SendMessageRequest, Message, ReceiveMessageRequest}
+
+object SQS {
+
+  /**
+   * SQS visibility timeout is the time window in which a message must be
+   * deleted (acknowledged). Otherwise it is considered abandoned.
+   * If a message has been pulled, but hasn't been deleted, the next time
+   * it will re-appear in another consumer is equal to `VisibilityTimeout`
+   * Another consequence is that if Loader has failed on a message processing,
+   * the next time it will get this (or anything) from a queue has this delay
+   * Specified in seconds
+   */
+  val VisibilityTimeout = 300
+
+  def mkClient[F[_]: Sync]: Resource[F, SqsClient] =
+    mkClientBuilder(identity[SqsClientBuilder])
+
+  def mkClientBuilder[F[_]: Sync](build: SqsClientBuilder => SqsClientBuilder): Resource[F, SqsClient] =
+    Resource.fromAutoCloseable(Sync[F].delay[SqsClient](build(SqsClient.builder()).build()))
+
+  def readQueue[F[_]: Timer: Sync](queueName: String): Stream[F, (Message, F[Unit])] = {
+
+    def getRequest(queueUrl: String) =
+      ReceiveMessageRequest
+        .builder()
+        .queueUrl(queueUrl)
+        .waitTimeSeconds(1)
+        .visibilityTimeout(VisibilityTimeout)
+        .build()
+
+
+    Stream.resource(mkClient.evalMap(getUrl[F](queueName))).flatMap { case (client, queueUrl) =>
+      Stream
+        .awakeEvery[F](1.second)
+        .flatMap { _ =>
+          val messages = Sync[F].delay(client.receiveMessage(getRequest(queueUrl))).map(_.messages().asScala)
```

Comment:
Is `client.receiveMessage` blocking or non-blocking?  If blocking, did you consider running it on a `Blocker`?

Reply:
It's blocking. But this function is used only in Loader itself and Loader has absolutely no throughput/performance requirements - I tried to keep it as simple as possible and left `Blocker` only in places where it was required by a function signature (mostly everything with doobie).